numba/tests/test_np_functions.py

```diff
@@ -588,6 +592,27 @@ def test_array_equal_exception(self):
             str(raises.exception)
         )
 
+    def test_intersect1d(self):
+
+        def arrays():
+            yield np.array([]), np.array([])  # two empty arrays
+            yield np.array([1]), np.array([])  # empty right
+            yield np.array([]), np.array([1])  # empty left
+            yield np.array([1]), np.array([2])  # singletons no intersection
+            yield np.array([1]), np.array([1])  # singletons one intersection
+            yield np.array([1, 2]), np.array([1])
+            yield np.array([1, 2, 2]), np.array([2, 2])
+            yield np.array([1, 2]), np.array([2, 1])
+            yield np.array([1, 2, 3]), np.array([1, 2, 3])
+
+        pyfunc = intersect1d
+        cfunc = jit(nopython=True)(pyfunc)
+
+        for a, b in arrays():
+            expected = pyfunc(a, b)
```

Comment:
Maybe you could simplify this test by casting to Numpy array here, rather than above in the test case generator?

Reply:
That is definitely better, I have simplified the test. Thank you